### PR TITLE
Releasing 2.2.4

### PR DESCRIPTION
**Pull request by unho**

### CODE REVIEW

File: docs/releases/2.2.4.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+*Released on 31 July 2017*
+
+This release contains improvements and bug fixes.
+
+
+Major changes
+=============
+
+- Full support for `.xliff` extension.
```
Comment (dwaynebailey) [inline docs/releases/2.2.4.rst]: Support for .xliff extension

Full support sounds like we're refering the XLIFF itself.

Comment (unho) [inline docs/releases/2.2.4.rst]: Will change.

File: docs/releases/2.2.4.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- Added support for nested and WebExtension JSON dialects.
+- `po2txt` no longer converts non-translatable strings.
```
Comment (dwaynebailey) [inline docs/releases/2.2.4.rst]: I think what that fixes actually does is use en if translation does not exist.

File: docs/releases/2.2.4.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
+Translate Toolkit 2.2.4
+***********************
+
+*Released on 31 July 2017*
+
+This release contains improvements and bug fixes.
+
+
+Major changes
```
Comment (dwaynebailey) [inline docs/releases/2.2.4.rst]: I think previous notes dropped Major.  These are not major changes.  They're just 'Summary of Changes'

File: docs/releases/2.2.4.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    extension.
+
+- JSON
+
+  - Added support for nested JSON.
+  - Added support for WebExtension JSON dialect.
+
+- txt
+
+  - `po2txt` skips obsolete and non-translatable strings.
```
Comment (dwaynebailey) [inline docs/releases/2.2.4.rst]: Again if asked to skip them it uses en instead fo blank.  At least that's what I think my change did :)

File: docs/releases/2.2.4.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+------------------
+
+- The `puncspace` check now strips Bidi markers chars before processing.
+- Added `ReducedChecker` checker to list of checkers.
+
+
+Tools
+-----
+
+- Added support for `.xliff` extension in all tools that support `.xlf` one.
```
Comment (dwaynebailey) [inline docs/releases/2.2.4.rst]: This has appeared now 3 times, put it in one place only in the details.

File: docs/releases/2.2.4.rst [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+Tools
+-----
+
+- Added support for `.xliff` extension in all tools that support `.xlf` one. 
+
+
+API changes
+-----------
+
+- Language and country default to `common_name` if available.
```
Comment (dwaynebailey) [inline docs/releases/2.2.4.rst]: I'm not sure this is an API change, maybe you disagree

**Comment by dwaynebailey** [comment]
Quite a few nits, otherwise gtm

File: docs/releases/2.2.4.rst (hunk 6 above)
Comment (unho) [inline docs/releases/2.2.4.rst]: I see no better place to put this change. I will leave it here.

File: docs/releases/2.2.4.rst (hunk 4 above)
Comment (unho) [inline docs/releases/2.2.4.rst]: No, my description is correct according to the commit messages, the code changes and the tests you landed. In other words, `po2txt` now correctly skips header and obsolete units from PO files when converting back to txt format.

**Pull request merged**